rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -465,16 +465,20 @@ class ReaderInfo : public ReaderListener
     bool trigger = false;
     mapmutex.lock();
     if (change->kind == ALIVE) {
-      topicNtypes[fqdn].insert(proxyData.typeName());
+      topicNtypes[fqdn].push_back(proxyData.typeName());
       trigger = true;
     } else {
       auto it = topicNtypes.find(fqdn);
-      if (
-        it != topicNtypes.end() &&
-        it->second.find(proxyData.typeName()) != it->second.end())
-      {
-        topicNtypes[fqdn].erase(proxyData.typeName());
-        trigger = true;
+      auto vector_find = [] (const auto & v, const auto & data) {
+          auto ret = std::find(std::begin(v), std::end(v), data);
+          return ret;
+        };
```

Comment:
Since you only use the lambda once I don't think that the overhead in complexity and reduced readability is justified. Please keep it as a simple if condition.

Same below.